src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1324,6 +1325,13 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
         }
 
         environment.put(GIT_COMMIT, revToBuild.revision.getSha1String());
+        if (launcher.isUnix()) {
+            String homeValue = environment.get("HOME");
+            environment.put(GIT_CEILING_DIRECTORIES, homeValue + "/jobs");
```

Comment:
I'm reasonably confident that Jenkins workspaces are allowed in many different locations.  The environment variable `HOME` (or `HOMEDRIVE` + `HOMEPATH` on Windows) is not necessarily the correct location.  I expect there are global Jenkins API's that will return the parent directory of the current workspace or the parent directory of all workspaces.
```suggestion
            environment.put(GIT_CEILING_DIRECTORIES, homeValue + "/jobs");
```